transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.file;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mule.util.FileUtils.openDirectory;
+
+import org.mule.api.exception.RollbackSourceCallback;
+import org.mule.api.exception.SystemExceptionHandler;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.util.concurrent.Latch;
+
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class FilePollEmptyResultTestCase extends FunctionalTestCase
+{
+
+    private static long TEST_TIMEOUT = 5000;
+
+    private static Throwable concurrentException;
+
+    private Latch latch = new Latch();
+
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "file-poll-empty-config.xml";
+    }
+
+    @Before
+    public void setUp() throws Exception
+    {
+        openDirectory(getFileInsideWorkingDirectory("test").getAbsolutePath());
+        openDirectory(getFileInsideWorkingDirectory("test2").getAbsolutePath());
+        muleContext.setExceptionListener(new TestExceptionHandler());
+    }
+
+    @Test
+    public void testNoConcurrentExceptionWhenEmptyPollResult() throws InterruptedException
+    {
+        assertThat(latch.await(TEST_TIMEOUT, MILLISECONDS), is(false));
```

Comment:
Is the test broken when the fix is not there?

Reply:
Yes. The test breaks consistently without the changes.